Apps/WebClient/src/ClientApp/src/components/timeline/encounter.vue

```diff
@@ -0,0 +1,193 @@
+                            </b-btn>
+                        </div>
+                        <b-collapse
+                            :id="'entryDetails-' + index + '-' + datekey"
+                            v-model="detailsVisible"
+                        >
+                            <div>
+                                <div class="detailSection">
+                                    <div>
+                                        <strong>Location:</strong>
+                                    </div>
+                                    <div>
+                                        {{ entry.clinic.name }}
+                                    </div>
+                                    <div>
+                                        {{ entry.clinic.address }}
+                                    </div>
+                                    <div>
+                                        {{
+                                            formatPhoneNumber(
+                                                entry.clinic.phoneNumber
+                                            )
+                                        }}
+                                    </div>
+                                </div>
+                            </div>
+                        </b-collapse>
+                    </b-col>
+                </b-row>
+            </b-col>
+        </b-row>
+        <CommentSection :parent-entry="entry"></CommentSection>
+    </b-col>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { Component, Prop, Ref } from "vue-property-decorator";
+import { Action, Getter, State } from "vuex-class";
+import { IconDefinition, faUserMd } from "@fortawesome/free-solid-svg-icons";
+import CommentSectionComponent from "@/components/timeline/commentSection.vue";
+import container from "@/plugins/inversify.config";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import User from "@/models/user";
+import moment from "moment";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import EncounterTimelineEntry from "@/models/encounterTimelineEntry";
+
+@Component({
+    components: {
+        CommentSection: CommentSectionComponent,
+    },
+})
+export default class EncounterTimelineEntryComponent extends Vue {
+    @Prop() entry!: EncounterTimelineEntry;
+    @Prop() index!: number;
+    @Prop() datekey!: string;
+    @Getter("user", { namespace: "user" }) user!: User;
+
+    private hasErrors: boolean = false;
+    private detailsVisible: boolean = false;
+
+    private mounted() {
+        // this.encounterService = container.get<IEncounterService>(
+        //     SERVICE_IDENTIFIER.EncounterService
+        // );
+    }
+
+    private get entryIcon(): IconDefinition {
+        return faUserMd;
+    }
+
+    private toggleDetails(): void {
+        this.detailsVisible = !this.detailsVisible;
+        this.hasErrors = false;
+    }
+
+    private formatPhoneNumber(phoneNumber: string): string {
```

Comment:
in overall, this seems to be a global util method as it is currently being used in many places i.e. medication.vue, verifySMS.vue and here

Reply:
Right - where would it live globally though?

Reply:
my bad for not providing any suggestions for this. I'm thinking about adding a phoneUtil.ts under utility?